test/gtest/ucp/test_ucp_perf.cc

```diff
@@ -187,16 +188,53 @@ UCS_TEST_P(test_ucp_perf, envelope) {
     ucs::scoped_setenv warn_invalid("UCX_WARN_INVALID_CONFIG", "no");
 
     /* Run all tests */
-    for (const test_spec *test_iter = tests; test_iter->title != NULL; ++test_iter) {
+    for (const test_spec *test_iter = tests; test_iter->title != NULL;
+         ++test_iter) {
         test_spec test = *test_iter;
 
-        if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_ARM_AARCH64) {
-            test.max *= UCP_ARM_PERF_TEST_MULTIPLIER;
-            test.min /= UCP_ARM_PERF_TEST_MULTIPLIER;
-        }
         test.iters = ucs_min(test.iters, max_iter);
+        test_adjust(test);
         run_test(test, 0, check_perf, "", "");
     }
 }
 
+
+UCS_TEST_P(test_ucp_perf, ucp_wait_mem) {
+    double perf_avg  = 0;
+    double perf_iter = 0;
+    int max_iter     = ucs_max(ucs::perf_retry_count, 1);
+    test_spec test;
+    int i;
+
+    /* No need to check with all transports - shm should be enough */
+    if (has_transport("shm")) {
```

Comment:
do UCP_INSTANTIATE_TEST_CASE only on shm transport

Reply:
How can it be done without impacting perf test ?

Reply:
we can create another test class (test_ucp_perf_wait_mem), inherit it from current test class (no additional members/methods), and instantiate it separately only for shm

Reply:
addressed